apiserver/apiserver.go

```diff
@@ -359,6 +359,7 @@ func (srv *Server) newTLSConfig(cfg ServerConfig) *tls.Config {
 		Cache:      srv.statePool.SystemState().AutocertCache(),
 		HostPolicy: autocert.HostWhitelist(cfg.AutocertDNSName),
 	}
+	go http.ListenAndServe(":http", m.HTTPHandler(nil))
```

Comment:
It seems very odd to me that a method called "newTLSConfig" would start a network listener.
Particularly as it might fail (perhaps port 80 is in use).

Reply:
Yeah, this was mostly a leftover from the initial prototype. I pushed a new commit with a possible reorganization of this part.